colcon_bundle/verb/bundlefile.py

```diff
@@ -28,7 +28,11 @@ def __init__(self, *, name=None, mode='w'):
         :param name: path to the bundle archive
         :param mode: mode to open file as
         """
-        self.tarfile = tarfile.open(name, mode)
+        with _set_temporary_tarfile_default_format(tarfile.GNU_FORMAT):
+            self.tarfile = tarfile.TarFile(
+                name=name, mode=mode, format=tarfile.DEFAULT_FORMAT)
+            tarfile.open(name, mode)
```

Comment:
Why are we losing the object that `open` is returning here? Written this way `open` is a no-op, and we always use an "empty" tarball, no?

Reply:
You're right, it's a no-op but we won't be using empty tarballs. From the `tarfile` documentation, [`tarfile.open()`](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.open) is an alternative constructor. Note that `open()` does not allow setting the format for the tarfile object.

[`tarfile.Tarfile()`](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile) recommends not using this method directly, instead using `open()` to create tarfile objects, but I don't see how we can set the format in that case.

Reply:
As you can see https://github.com/python/cpython/blob/3.8/Lib/tarfile.py#L1551 `tarfile.open` is passing down `**kwargs` to `tarfile.open` so you can just pass `format` there I think?

Reply:
Done!